### PR TITLE
feat: health checks for Google Cloud Config Connector resources

### PR DESCRIPTION
**Pull request by Kerwood**
Signed-off-by: Patrick Kerwood <patrick@kerwood.dk>

Added health checks for the following Google Cloud Config Connector resouces.

- sql.cnrm.cloud.google.com
  - SQLInstance
  - SQLDatabase
  - SQLUser
- storage.cnrm.cloud.google.com
  - StorageBucket
- iam.cnrm.cloud.google.com
  - IAMServiceAccount
  - IAMPartialPolicy
  - IAMPolicyMember (existed but refactored to match the others resources)

Edit:
Since Config Connector uses the same status indication on all resources, the health checks are the same, only difference is the `apiVersion` and `kind` in the test objects.

### CODE REVIEW

File: resource_customizations/iam.cnrm.cloud.google.com/IAMPolicyMember/testdata/healthy_upToDate.yaml [hunk 1]
```diff
@@ -1,77 +0,0 @@
-apiVersion: iam.cnrm.cloud.google.com/v1beta1
```
Comment (crenshaw-dev) [inline resource_customizations/iam.cnrm.cloud.google.com/IAMPolicyMember/testdata/healthy_upToDate.yaml:1]: Looks like the contents were deleted, but the file was not. Is that the intention?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: resource_customizations/iam.cnrm.cloud.google.com/IAMPolicyMember/testdata/healthy_upToDate.yaml (hunk 1 above)
Comment (Kerwood) [reply resource_customizations/iam.cnrm.cloud.google.com/IAMPolicyMember/testdata/healthy_upToDate.yaml:1]: Not sure what you mean? The file `iam.cnrm.cloud.google.com/IAMPolicyMember/testdata/healthy_upToDate.yaml` does not exist.
https://github.com/Kerwood/argo-cd/tree/health-check/google-cloud/resource_customizations/iam.cnrm.cloud.google.com/IAMPolicyMember/testdata

Comment (crenshaw-dev) [reply resource_customizations/iam.cnrm.cloud.google.com/IAMPolicyMember/testdata/healthy_upToDate.yaml:1]: Ah it's just GitHub's rendering confusing me then. Thanks!

**Review by crenshaw-dev — COMMENTED** [review]
@Kerwood my only hesitation to merge is that it's not super clear whether the IAMPolicyMember health check behaves the same as the old one. Would it be possible to leave the old tests in place, just to make it extra clear?

**Comment by crenshaw-dev** [comment]
Ah, no worries then. Just wanted to make sure there weren't changes to the health check behavior. Thanks for clarifying!

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**